test/string.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <unity.h>
+
+static void
+test_Strcmp(void)
+{
+    /* Test with long string. */
+    TEST_ASSERT_EQUAL(0, strcmp("Here is a string", "Here is a string"));
+
+    /* Test with medium string. */
+    TEST_ASSERT_EQUAL(0, strcmp("Hello World", "Hello World"));
+
+    /* Test with short string. */
+    TEST_ASSERT_EQUAL(0, strcmp("A", "A"));
+
+    /* Test with numbers in string. */
+    TEST_ASSERT_EQUAL(0, strcmp("He11o W0rld", "He11o W0rld"));
+
+    /* Test with special characters. */
+    TEST_ASSERT_EQUAL(0, strcmp("Hello_World!", "Hello_World!"));
+
+    /* Test string with only a space. */
+    TEST_ASSERT_EQUAL(0, strcmp(" ", " "));
+
+    /* Test an empty string. */
+    TEST_ASSERT_EQUAL(0, strcmp("", ""));
+
+    /* Test two different strings. */
+    int result = strcmp("Here is a string", "Here is a different string");
+    TEST_ASSERT(result != 0);
+}
+
+static void
+test_Strlen(void)
+{
+    /* Test with long string. */
+    TEST_ASSERT_EQUAL(59, strlen("Here is a long string. "
+                                 "This goes above thirty-two charcters"));
+
+    /* Test with medium string. */
+    TEST_ASSERT_EQUAL(23, strlen("Here is a medium string"));
+
+    /* Test with short string. */
+    TEST_ASSERT_EQUAL(6, strlen("String"));
+
+    /* Test with numbers in string. */
+    TEST_ASSERT_EQUAL(7, strlen("String7"));
+
+    /* Test with special characters. */
+    TEST_ASSERT_EQUAL(8, strlen("Str-ing!"));
+
+    /* Test with empty string. */
+    TEST_ASSERT_EQUAL(0, strlen(""));
+}
+
+static void
+test_Memcpy(void)
```

Comment:
Like I mentioned earlier, this should test more than just strings (e.g. integers of various sizes).

Reply:
done